scipy/special/stirling2.h

```diff
@@ -40,7 +43,11 @@ double _stirling2_dp(double n, double k){
         return 0.;
     }
     int arraySize = k <= n - k + 1 ? k : n - k + 1;
-    double *curr = (double *) malloc(arraySize * sizeof(double));
+    auto curr = std::unique_ptr<double[]>{new (std::nothrow) double[arraySize]()};
```

Comment:
```suggestion
    auto curr = std::unique_ptr<double[]>{new (std::nothrow) double[arraySize]};
```
Those parentheses cause the array to be initialized with zeros.  This is not needed, since `curr` will be initialized to 1 below.